libyul/backends/evm/ControlFlowGraphBuilder.cpp

```diff
@@ -628,3 +687,32 @@ void ControlFlowGraphBuilder::jump(
 	_target.entries.emplace_back(m_currentBlock);
 	m_currentBlock = &_target;
 }
+
+void ControlFlowGraphBuilder::makeSwitch(
+	shared_ptr<DebugData const> _debugData,
+	StackSlot _switchExpr,
+	CFG::BasicBlock* defaultCase,
+	std::map<u256, CFG::BasicBlock*> cases,
+	CFG::BasicBlock& target
+)
+{
+	yulAssert(m_currentBlock, "");
+	m_currentBlock->exit = CFG::BasicBlock::Switch{
+		move(_debugData),
+		move(_switchExpr),
+		(defaultCase != nullptr) ? defaultCase : &target,
+		cases,
+	};
+	if (defaultCase == nullptr)
+	{
+		target.entries.emplace_back(m_currentBlock);
+		m_currentBlock = &target;
+	}
+	else
+	{
+		defaultCase->entries.emplace_back(m_currentBlock);
+		m_currentBlock = nullptr;//&target;
+	}
+	for (auto const& [caseVal, caseBlock]: cases)
+		caseBlock->entries.emplace_back(m_currentBlock);
+}
```

Comment:
```suggestion
	if (defaultCase == nullptr)
		target.entries.emplace_back(m_currentBlock);
	else
		defaultCase->entries.emplace_back(m_currentBlock);
	for (auto const& [caseVal, caseBlock]: cases)
		caseBlock->entries.emplace_back(m_currentBlock);
	m_currentBlock = nullptr;//&target;
}
```

This is the last thing I saw: all cases need entries from the *switch* block, but you've overwritten ``m_currentBlock`` (not sure why) above, so it's no longer the switch block here.